middlewares/secrets.go

```diff
@@ -0,0 +1,32 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+type SecretsConfig struct {
+	OmitHeaders []string `yaml:"omit_headers"` // Headers to be omitted for secrets protection
+}
+
+// Secrets middleware removes specified headers from the request and response to enhance security.
+func NewSecretsMiddleware(config SecretsConfig) func(http.Handler) http.Handler {
```

Comment:
Do not use config in middleware, pass the list of headers

Reply:
And more specific name for the middleware like "OmitHeadersMiddleware"

Reply:
Fixed